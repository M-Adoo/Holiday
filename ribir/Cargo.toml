[package]
authors.workspace = true
categories.workspace = true
description.workspace = true
documentation.workspace = true
edition.workspace = true
homepage.workspace = true
keywords.workspace = true
license.workspace = true
name = "ribir"
readme.workspace = true
repository = "https://github.com/RibirX/Ribir/ribir"
version.workspace = true

[dependencies]
lyon_algorithms.workspace = true
lyon_path.workspace = true
ribir_core = {path = "../core", version = "0.0.0", features = ["png"]}
ribir_gpu = {path = "../gpu", version = "0.0.0"}
ribir_widgets = {path = "../widgets", version = "0.0.0"}

[dev-dependencies]
env_logger.workspace = true
png.workspace = true

[features]
default = ["wgpu_gl"]
wgpu_gl = ["ribir_gpu/wgpu_gl"]
