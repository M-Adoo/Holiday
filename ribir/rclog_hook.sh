#!/bin/bash

if [ "$MERGE_CHANGELOG" = true ]; then
  echo "$(rclog -t $NEW_VERSION -p ./CHANGELOG.md merge)" >| ./CHANGELOG.md
fi
