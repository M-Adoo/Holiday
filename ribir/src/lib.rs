pub use ribir_core as core;
pub use ribir_widgets as widgets;
pub mod prelude {
  pub use ribir_core::prelude::*;
  pub use ribir_widgets::prelude::*;
  pub mod app {

    #[cfg(feature = "wgpu_gl")]
    pub fn run(root: super::Widget) {
      let mut app = super::Application::default();
      let wnd = app.new_window(|native_wnd, ctx| {
        let size = native_wnd.inner_size();
        let p_backend = super::AppContext::wait_future(gpu::wgpu_backend_with_wnd(
          &native_wnd,
          super::DeviceSize::new(size.width, size.height),
          None,
          None,
          ctx.shaper.clone(),
        ));
        super::Window::new(native_wnd, p_backend, root.into_widget(), ctx)
      });
      app.exec(wnd);
    }
  }
}

use prelude::*;
#[cfg(feature = "wgpu_gl")]
pub fn wgpu_headless_wnd(root: Widget, ctx: AppContext, size: DeviceSize) -> Window {
  let p_backend = AppContext::wait_future(gpu::wgpu_backend_headless(
    size,
    None,
    None,
    ctx.shaper.clone(),
  ));
  Window::new(
    ribir_core::window::MockRawWindow {
      size: Size::from_untyped(size.to_f32().to_untyped()),
      ..Default::default()
    },
    p_backend,
    root,
    ctx,
  )
}
