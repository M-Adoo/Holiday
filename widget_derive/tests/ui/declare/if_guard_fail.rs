use ribir::prelude::*;

fn normal_if_guard_pass() {
  let guard = Some(1);
  let _ = declare! {
    SizedBox {
      // if guard in widget's field
      size if true => : Size::zero(),
      // if guard in data atribute
      cursor if true => : CursorIcon::Hand,
      // if guard in listener attibute
      on_tap if let Some(_) = guard  => : |_| {},
      // if guard in wrap widget.
      margin if true => : EdgeInsets::all(1.)
    }
  };
}

fn id_if_guard_fail() {
  let guard = Some(1);
  let _ = declare! {
    SizedBox {
      id if true => : test,
      // if guard in widget's field
      size if true => : Size::zero(),

    }
  };
}

fn depend_id_behind_if_guard_fail() {
  let _ = declare! {
    SizedBox {
      id: a,
      size: Size::zero(),
      margin if true =>:  EdgeInsets::all(0.),

      SizedBox{
        size: Size::zero(),
        margin: a.margin
      }

    }
  };
}

fn main() {}
