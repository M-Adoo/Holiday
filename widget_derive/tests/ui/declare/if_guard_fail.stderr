error: if guard is not supported in `id`
  --> $DIR/if_guard_fail.rs:23:10
   |
23 |       id if true => : test,
   |          ^^^^^^^

error: Depends on a field which behind `if guard`, its existence depends on the `if guard` result in runtime.
   |
help: field define here.
  --> $DIR/if_guard_fail.rs:34:11
   |
34 |         id: a,
   |  ___________^
35 | |       size: Size::zero(),
36 | |       margin if true =>:  EdgeInsets::all(0.),
   | |____________^ ^^^^^^^
